zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/Collection.java

```diff
@@ -1173,7 +1173,11 @@ public boolean deleteDataVisContent(Session session, Path contentPath) throws IO
             return false;
         }
 
-        String visualisationZipUri = contentPath.toString();
+        String visualisationZipUri = contentPath.getParent().toString();
+        if (visualisationZipUri == null || StringUtils.isEmpty(visualisationZipUri)){
```

Comment:
[`StringUtils.isEmpty` ](http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isEmpty-java.lang.CharSequence-) will handle null cases so you can remove the `visualisationZipUri == null` check.

Reply:
you're right! I was following the pattern I saw in other checks, but it does seem redundant